### PR TITLE
feat: mates section

### LINKED ISSUE #387: [FEATURE] create a mates/users page
### Description

Requirements:

We already have an API built to get all users data (https://www.projectmate.net/api/user/all) 

We need to just display the users data in a creative way in https://www.projectmate.net/mates page. 
Please create a PR for the same and let us know if you need anything. Thanks

### PR DESCRIPTION
**Pull request by AryanKhubchandani**
### Fixes Issue

Closes #387

### Changes proposed

- Fetch data from given api
- Add UI for mates section to show fetched data

### Screenshots
image

### CODE REVIEW

File: components/MatesList/MatesList.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+  if (error) {
+    return <div className="m-auto my-5 text-lg">Failed to load mates</div>;
+  }
+
+  return (
+    <section className="py-16 px-2">
+      <div className="mx-auto flex max-w-screen-xl flex-col gap-8">
+        <p className="text-center text-xl text-gray-200 md:text-3xl lg:text-4xl">
+          Mates
```
Comment (rohitdasu) [inline components/MatesList/MatesList.tsx:18]: don't need heading... this is already highlighted in the sidemenu

**Review by rohitdasu — CHANGES_REQUESTED** [review]

File: components/MatesList/MatesList.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import { Mate } from './Mate';
+import useSWR from 'swr';
+import { fetcher } from '@/lib/fetcher';
+
+export const MatesList = () => {
+  const url = `https://www.projectmate.net/api/user/all`;
+  const { data, error } = useSWR(url, fetcher);
```
Comment (JakubChorzepa) [inline components/MatesList/MatesList.tsx:8]: This is not proper way of passing url query string from internal next service.
[_Any file inside the folder pages/api is mapped to /api/*_](https://nextjs.org/docs/pages/building-your-application/routing/api-routes).
Please use url string like this `/api/user/all`

**Review by JakubChorzepa — CHANGES_REQUESTED** [review]

**Review by rohitdasu — CHANGES_REQUESTED** [review]
hi, everything looks fine but could you add a loading state in the UI?

File: components/MatesList/MatesList.tsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import { Mate } from './Mate';
+import useSWR from 'swr';
+import { fetcher } from '@/lib/fetcher';
+
+export const MatesList = () => {
+  const url = `/api/user/all`;
+  const { data, error } = useSWR(url, fetcher);
```
Comment (rohitdasu) [inline components/MatesList/MatesList.tsx:8]: here you can get loading state... just display plain text stating 'loading'. later we can improve it

File: components/MatesList/Mate.tsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  return (
+    <div>
+      <li
+        className="flex flex-col items-center justify-center gap-2"
+        key={mate.id}
+      >
+        <Image
+          src={
+            mate.profilePicture ||
+            `https://avatars.dicebear.com/api/initials/${mate.name}.png?backgroundColorLevel=800&fontSize=40`
```
Comment (rohitdasu) [inline components/MatesList/Mate.tsx:14]: can you put it in a function and use it here just to make more readable?

**Review by JakubChorzepa — APPROVED** [review]
LGTM :+1:

**Review by rohitdasu — APPROVED** [review]
LGTM

**Pull request merged**